From 0d64bbddaa0078ef148d609a3cfad854cf00d7de Mon Sep 17 00:00:00 2001
From: Martin Pitt <martinpitt@gnome.org>
Date: Fri, 8 Nov 2013 14:59:50 +0100
Subject: [PATCH] lib: Fix segfault on getting property when daemon is not
 running

This fixes "upower --version" when the daemon is not running, and thus the
client proxy is NULL.
---
 libupower-glib/up-client.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libupower-glib/up-client.c b/libupower-glib/up-client.c
index 35d7b5d..17fb02d 100644
--- a/libupower-glib/up-client.c
+++ b/libupower-glib/up-client.c
@@ -322,6 +322,9 @@ up_client_get_property (GObject *object,
 	UpClient *client;
 	client = UP_CLIENT (object);
 
+	if (client->priv->proxy == NULL)
+                return;
+
 	switch (prop_id) {
 	case PROP_DAEMON_VERSION:
 		g_value_set_string (value, up_client_glue_get_daemon_version (client->priv->proxy));
-- 
1.8.4.2

